### PR TITLE
Ft database integration 163136994

### PR DESCRIPTION
**Pull request by kossy360**
#### What does this PR do?
 - Integrate a database for data persistence
 - Add extra endpoints
 - Secures all routes with JWT
#### Description of Task to be completed?
 - build a database with postgresql
 - have the following endpoints working
```
POST /api/v1/auth/signup
```
```
POST /api/v1/auth/login
```
```
POST /api/v1/comments
```
```
GET /api/v1/comments/<question-id>
```
```
POST /api/v1/notifications/<meetup-id>/register
```
```
GET /api/v1/notifications
```
```
PATCH /api/v1/meetups/<meetup-id>
```
```
PATCH /api/v1/users/update
```
```
PATCH /api/v1/notifications/<meetup-id>/reset
```
```
DELETE /api/v1/meetups/<meetup-id>
```
```
DELETE /api/v1/notifications/<meetup-id>/clear
```
- Secure all access routes with JWT
#### How should this be manually tested?
 - carry out all manual tests using POSTMAN, you should provide the JWT token in the 'auth' header of all requests
 - JWTs are sent as part of the response body after successful signup or login
 - All routes that create, modify or delete a meetup all require admin priviledges
 - to carry out automated tests using jasmine, you need to provide necessary database credentials in a .env file.
#### What are the relevant pivotal tracker stories?
#163136994
#163240264
#163240489
#163240749
#163241034
#163241248
#163241363
#163241514
#163243649
#163243692
#163243749
#163254688

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// import storage from '../storage';
```
Comment (tosmak16) [inline controllers/users.js]: @kossy360 remove commented codes in line 1

File: middleware/authenticator.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import jwt from 'jsonwebtoken';
+import error from './errorhandler';
+
+const authenticator = {
+  generate: body => new Promise((res) => {
```
Comment (tosmak16) [inline middleware/authenticator.js]: @kossy360 Change generate to generateToken

File: controllers/users.js [hunk 3]
```diff
@@ -36,6 +36,28 @@ const controller = {
       else error(500, res);
     }
   },
+
+  getUser: async (req, res) => {
```
Comment (tosmak16) [inline controllers/users.js]: @kossy360 Well done! Use a more descriptive name.

File: middleware/validator.js [hunk 4]
```diff
@@ -1,28 +1,70 @@
...
+    othername: joi.string().alphanum().min(1),
+    email: joi.string().email(),
+    phonenumber: joi.string().regex(/\d/).min(6).max(15),
+    username: joi.string().alphanum().min(3).max(10),
+  }),
+
   meetup: joi.object().keys({
-    createdOn: joi.number().min(Date.now()).required(),
-    happeningOn: joi.number().min(Date.now()).required(),
+    happening: joi.number().integer().min(Date.now()).required(),
```
Comment (tosmak16) [inline middleware/validator.js]: Hello @kossy360 I think `happening` in line 32 should be `happeningOn`

File: server/controllers/questions.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  const val = vote.toLowerCase();
+  if (val === 'upvote') return 1;
+  if (val === 'downvote') return -1;
+  return 0;
+};
+
+const control = {
+  getAll: async (req, res) => {
+    try {
+      const { meetupId } = await validator(req.params, 'reqId');
```
Comment (tosmak16) [inline server/controllers/questions.js]: Hello, @kossy360 What is `reqId`?

File: server/db/databaseSQL.js [hunk 6]
```diff
@@ -0,0 +1,376 @@
+const test = `SET statement_timeout = 0;
```
Comment (tosmak16) [inline server/db/databaseSQL.js]: Hello, Well done @kossy360 Please change `test` to a more descriptive name.

File: server/controllers/questions.js (hunk 5 above)
Comment (kossy360) [reply server/controllers/questions.js]: it's the key of the Joi schema object that validates all request parameters

Comment (tosmak16) [reply server/controllers/questions.js]: Okay.

**Comment by tosmak16** [comment]
@kossy360 You can merge this PR.

**Pull request merged**